src/psyclone/psyir/frontend/fortran.py

```diff
@@ -55,14 +55,18 @@ class FortranReader():
     ''' PSyIR Fortran frontend. This frontend translates Fortran from a string
     or a file into PSyIR using the fparser2 utilities.
 
+    :param modules_to_resolve: Optional list of modules names that when found
+        the frontend will attempt to resolve their symbol information.
+    :rtype: list[str] | None
+
     '''
     # Save parser object across instances to reduce the initialisation time
     _parser = None
 
-    def __init__(self):
+    def __init__(self, modules_to_resolve=None):
         if not self._parser:
             self._parser = ParserFactory().create(std="f2008")
```

Comment:
I now added validation in the Fparser2Reader constructor, since this parameter is just passed to that class and not stored here, is this enough? or should I repeat the validation here?